crates/core/tedge/src/cli/mqtt/cli.rs

```diff
@@ -32,7 +36,8 @@ pub enum TEdgeMqttCli {
     /// Subscribe a MQTT topic.
     Sub {
         /// Topic to subscribe to
-        topic: String,
+        #[arg(value_parser = TopicFilter::new)]
```

Comment:
```suggestion
        #[arg(value_parser = SimpleTopicFilter::new)]
```
Without that, I didn't really  get how the implicit conversion from `TopicFilter` into `SimpleTopicFilter` was happening.

Reply:
Yup ... leading to tons of failing messages. => fixed.

What's weird is that this compile fine.

Reply:
> What's weird is that this compile fine.

Yeah, I was also surprised by that.